src/components/forms/DataLocation.tsx

```diff
@@ -0,0 +1,84 @@
+import React, {useCallback} from 'react';
+import {FormField, FormFieldTypes, FormValidators} from './forms';
+import {cloneDeep} from 'lodash';
+import {set} from 'lodash';
+import {Link} from 'react-router-dom';
+import {Styles} from '../../libs/theme';
+import {asIdAndDisplayText, SelectEntry} from './SelectOptionInterface';
+import {CloudProviders} from './CloudProviders';
+import {DataLocations} from "./DataLocations";
+import {ResearchStage} from "./ResearchStage";
+
+export type DataLocationInfo = {
+    cloudProvider: null | SelectEntry,
+    dataLocation: null | SelectEntry,
+    locationUrl: null | string,
+    researchStage: null | SelectEntry,
+};
+
+export type DataLocationComponentProps = {
+    idx: number,
+    location: DataLocationInfo,
+    onChange: ({idx, location}: { idx: number, location: DataLocationInfo }) => void,
+    onDelete: (idx: number) => void,
+};
+
+export const DataLocation = (props: DataLocationComponentProps) => {
+    const {idx, location, onChange, onDelete} = props;
+
+    const onDataLocationChange = useCallback(({key, value}: { key: string, value: string | SelectEntry }) => {
+        const newDataLocation = cloneDeep(location);
+        set(newDataLocation, key, value)
+        onChange({idx: idx, location: newDataLocation});
+    }, [location, idx, onChange]);
+
+    const deleteButton = <Link
+        style={{marginLeft: '15px'}}
+        id={`${idx}_deleteDataLocation`}
+        className={'glyphicon glyphicon-trash'}
+        onClick={() => onDelete(idx)}
+        to={`#`}
+    />
+
+    return (<div style={Styles.REPEATING_SECTION}>
+        <h4>Data Location {idx + 1} {deleteButton}</h4>
+        <FormField
+            style={{width: '50%', paddingBottom: '10px'}}
```

Comment:
Given that all the `FormField` components have the same style, should we abstract it to a variable?

Reply:
Makes sense.  I'll extract to a variable.